src/test_trie.py

```diff
@@ -0,0 +1,196 @@
+"""Tests for Python implementation of Trie."""
+import pytest
+from trie import Trie, Node
+
+
+@pytest.fixture
+def init_empty_trie():
+    """Init empty trie."""
+    return Trie()
+
+
+@pytest.fixture
+def init_trie_one_word():
+    """Init empty trie."""
+    new_trie = Trie()
+    new_trie.insert('cake')
+    return new_trie
+
+
+@pytest.fixture
+def init_trie_three_words_shared_letters():
+    """Init empty trie."""
+    new_trie = Trie()
+    new_trie.insert('be')
+    new_trie.insert('bee')
+    new_trie.insert('bell')
+    return new_trie
+
+
+@pytest.fixture
+def init_jumbo_list():
+    """Fixture for 250k word test."""
+    # jumbo_trie = Trie()
+    word_file = open('/usr/share/dict/words', 'r')
+    word_text = word_file.read()
+    word_file.close()
+    word_text = word_text.split('\n')
+    # for word in word_text:
+    #     jumbo_trie.insert(word)
+    return word_text
+
+
+def test_init_empty_node():
+    """Test that node init attributes empty by default."""
+    new_node = Node()
+    assert new_node._children == []
+    assert new_node._data is None
+    assert new_node._parent is None
+
+
+def test_init_node_with_value_and_parent():
+    """Test that node init with parent and val works."""
+    parent_node = Node('c')
+    new_node = Node(parent_node, 'a')
+    assert new_node._parent == parent_node
+    assert new_node._data == 'a'
+
+
+def test_init_empty_trie_has_correct_attrs(init_empty_trie):
+    """Test an empty trie intance has correct attributes."""
+    root = init_empty_trie._root
+    assert type(root) == Node
+    assert root._data is None
+    assert init_empty_trie._size == 0
+
+
+def test_insert_iterable(init_empty_trie):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(TypeError):
+        init_empty_trie.insert(['string'])
+
+
+def test_insert_int(init_empty_trie):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(TypeError):
+        init_empty_trie.insert(300)
+
+
+def test_insert_two_words(init_empty_trie):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(ValueError):
+        init_empty_trie.insert('hi there')
+
+
+def test_insert_duplicate(init_trie_one_word):
+    """Check error is raised when iterable inserted."""
+    with pytest.raises(ValueError):
```

Comment:
I don't think you should explicitly raise an error. I think in this situation it should just do nothing, That way it won't stop a program from running if you insert the same word.